cmake/ext_imgcodecs.cmake

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+set(_IMGCODEC_ROOT_DIR ${dlib_SOURCE_DIR}/dlib/external)
+
+# ----------------------------------------------------------------------------
+#  project libpng
+#
+# ----------------------------------------------------------------------------
+set (PNG_LIBRARY "libpng_static_c")
+set (libPNG_SOURCE_DIR ${_IMGCODEC_ROOT_DIR}/libpng)
+set (zlib_SOURCE_DIR ${_IMGCODEC_ROOT_DIR}/zlib)
+
+if(NOT WIN32)
+  find_library(M_LIBRARY
+    NAMES m
+    PATHS /usr/lib /usr/local/lib
+  )
+  if(NOT M_LIBRARY)
+    message(STATUS "math lib 'libm' not found; floating point support disabled")
+  endif()
+else()
+  # not needed on windows
+  set(M_LIBRARY "")
+endif()
+
+set(lib_srcs
+   ${libPNG_SOURCE_DIR}/arm/arm_init.c
```

Comment:
is there a reason we include all these files individually rather than put them in folders and include the folders instead for efficiency? (same for libjpeg)

Reply:
Yes, there are some unnecessary source files in the folder, and cannot be compiled easily. e.g. a lot of build configuraiton and tunning.